No compilation errors are shown

I ran the following code:

```go
package go1a

// Describe takes an amount and returns a description.
func Describe(amount int) string {
	return ""
}

func Units(amount int) (int, int, int, int) {
	grosses := amount / 144
	amount %= 144
	
	scores := amount / 20
	amount %= 20
	
	dozens := amount / 12
	amount %= 12
	
	return (grosses, scores, dozens, amount)
}
```

This has a syntax error, but it didn't show up in the test output. Looking at the test results JSON, it has captured the compile error, but in the test itself and not in the top-level `message` property (see the [spec](https://github.com/exercism/automated-tests/blob/master/docs/interface.md#message)):

```json
{
	"status": "error",
	"tests": [
		{
			"name": "build",
			"status": "error",
			"expected": "",
			"actual": "",
			"message": "./go_1_a.go:18:17: syntax error: unexpected comma, expecting )"
		}
	]
}
```

For this reason., the compilation error message is not shown as the output. The test runner should output the compilation error in the top-level `message` property.

--- a/cmd/testoutput/main.go
+++ b/cmd/testoutput/main.go
@@ -3,6 +3,7 @@ package main
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io/ioutil"
 	"log"
@@ -53,7 +54,13 @@ func getStructure(lines [][]byte) *exreport.Report {
 		Tests:  nil,
 	}
 
-	tests := buildTests(lines)
+	tests, err := buildTests(lines)
+	if err != nil {
+		report.Status = statErr
+		report.Message = err.Error()
+		return report
+	}
+
 	for _, test := range tests {
 		if test == nil {
 			// just to be sure we dont get a nil pointer exception
@@ -75,13 +82,12 @@ func getStructure(lines [][]byte) *exreport.Report {
 	return report
 }
 
-func buildTests(lines [][]byte) map[string]*exreport.Test {
+func buildTests(lines [][]byte) (map[string]*exreport.Test, error) {
 	var (
 		tests   = map[string]*exreport.Test{}
 		failMsg [][]byte
 	)
 
-forLines:
 	for _, lineBytes := range lines {
 		var line testLine
 
@@ -92,12 +98,7 @@ forLines:
 			// if there is a failure running the tests, supress the line with `#` at the beginning
 			continue
 		case bytes.HasPrefix(lineBytes, []byte("FAIL")):
-			tests["build"] = &exreport.Test{
-				Name:    "build",
-				Status:  statErr,
-				Message: string(bytes.Join(failMsg, []byte{'\n'})),
-			}
-			break forLines
+			return nil, errors.New(string(bytes.Join(failMsg, []byte{'\n'})))
 		case !bytes.HasPrefix(lineBytes, []byte{'{'}):
 			// if the line is not a json, we need to collect the lines to gather why `go test --json` failed
 			failMsg = append(failMsg, lineBytes)
@@ -127,7 +128,7 @@ forLines:
 			tests[line.Test].Status = statPass
 		}
 	}
-	return tests
+	return tests, nil
 }
 
 type testLine struct {

--- a/cmd/testoutput/main.go
+++ b/cmd/testoutput/main.go
@@ -53,6 +53,11 @@ func getStructure(lines [][]byte) *exreport.Report {
 		Status: statPass,
 		Tests:  nil,
 	}
+	defer func() {
+		if report.Tests == nil {
+			report.Tests = []exreport.Test{}
+		}
+	}()
 
 	tests, err := buildTests(lines)
 	if err != nil {

--- a/exreport/report.go
+++ b/exreport/report.go
@@ -2,6 +2,7 @@ package exreport
 
 // Report represents a report of all tests of an exercise.
 type Report struct {
-	Status string `json:"status"`
-	Tests  []Test `json:"tests"`
+	Status  string `json:"status"`
+	Message string `json:"message,omitempty"`
+	Tests   []Test `json:"tests"`
 }

